front/src/modules/timesStops/helpers/computeMargins.tsx

```diff
@@ -0,0 +1,108 @@
+import type { SimulationResponseSuccess } from 'applications/operationalStudies/types';
+import type { TrainScheduleResult } from 'common/api/generatedEditoastApi';
+import type { OperationalPointWithTimeAndSpeed } from 'modules/trainschedule/components/DriverTrainScheduleV2/types';
+import { interpolateValue } from 'modules/trainschedule/components/DriverTrainScheduleV2/utils';
+import {
+  decimalToPercentage,
+  mToMm,
+  milliUnitToBaseUnit,
+  minutePer100km,
+  percentageToDecimal,
+  timeLostFromMin100KmMargin,
+} from 'utils/physics';
+import type { ValueOf } from 'utils/types';
+
+import { formatDigitsAndUnit } from './utils';
+import { MarginUnit, marginRegExValidation } from '../consts';
+
+// We use the previous unit used if it’s not given
+type MarginUnit = ValueOf<typeof MarginUnit>;
+let marginUnit: MarginUnit = MarginUnit.percent;
+
+const defaultReturn = {
+  theoreticalMarginWithUnit: '',
+  calculatedMargin: '',
+  diffMarginsWithUnit: '',
+};
+const computeMargins = (
+  trainSimulation: SimulationResponseSuccess,
+  opPoint: OperationalPointWithTimeAndSpeed,
+  nextOpPoint: OperationalPointWithTimeAndSpeed,
+  selectedTrainSchedule: TrainScheduleResult,
+  pathId: string
+) => {
+  const theoreticalMarginIndex = selectedTrainSchedule.margins?.boundaries?.findIndex(
+    (id) => id === pathId
+  );
+  if (
+    theoreticalMarginIndex === undefined ||
+    theoreticalMarginIndex < 0 ||
+    theoreticalMarginIndex > selectedTrainSchedule.margins!.values.length - 2
+  ) {
+    return defaultReturn;
+  }
+  const theoreticalMargin = selectedTrainSchedule.margins!.values[theoreticalMarginIndex + 1];
+  const theoriticalMarginRegexMatches = theoreticalMargin.match(marginRegExValidation);
+  if (!theoreticalMargin || theoreticalMargin === 'none' || !theoriticalMarginRegexMatches) {
+    return defaultReturn;
+  }
+  marginUnit = (theoriticalMarginRegexMatches[3] as MarginUnit) || marginUnit;
+  const theoreticalMarginValue = Number(theoriticalMarginRegexMatches[1]);
+
+  // base duration to go from current scheduled point to next scheduled point
+  const opPointBaseTime = interpolateValue(trainSimulation.base, mToMm(opPoint.position), 'times');
+  const nextOpPointBaseTime = interpolateValue(
+    trainSimulation.base,
+    mToMm(nextOpPoint.position),
+    'times'
+  );
+  const baseDuration = milliUnitToBaseUnit(nextOpPointBaseTime - opPointBaseTime);
```

Comment:
milliUnitToBaseUnit is a name quite abstract, I think msToS is clearer
```suggestion
  const baseDuration = msToS(nextOpPointBaseTime - opPointBaseTime);
```

Reply:
done